libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -231,10 +228,39 @@ - (void)send:(SFRestRequest *)request requestDelegate:(nullable id<SFRestRequest
     [self send:request requestDelegate:requestDelegate shouldRetry:self.requiresAuthentication && request.requiresAuthentication];
 }
 
-- (void)send:(SFRestRequest *)request requestDelegate:(id<SFRestRequestDelegate>)requestDelegate shouldRetry:(BOOL)shouldRetry {
+- (void)send:(SFRestRequest *)request requestDelegate:(nullable id<SFRestRequestDelegate>)requestDelegate shouldRetry:(BOOL)shouldRetry {
     if (requestDelegate != nil) {
         request.requestDelegate = requestDelegate;
     }
+
+    [self send:request failureBlock:^(id  _Nullable response, NSError * _Nullable e, NSURLResponse * _Nullable rawResponse) {
+        [self notifyDelegateOfFailure:requestDelegate
+                              request:request
+                                 data:response
+                          rawResponse:rawResponse
+                                error:e];
+    } successBlock:^(id  _Nullable response, NSURLResponse * _Nullable rawResponse) {
+        [self notifyDelegateOfSuccess:requestDelegate
+                              request:request
+                                 data:response
+                          rawResponse:rawResponse];
+    } shouldRetry: shouldRetry];
+}
```

Comment:
💬 This change bridges the legacy delegate-based request path with the new block-based pipeline.

Previously, block-based requests were injected into the delegate system at runtime using objc_setAssociatedObject, which was fragile and hard to reason about. Now, everything below this point uses native block handling without relying on Objective-C runtime hacks.

This method preserves the public delegate API but internally forwards to the new unified block-based implementation, allowing us to consolidate logic, improve maintainability, and ensure consistent behavior across both patterns.

See the related diff for the full context on the transition from SFRestAPI+Blocks.m to Swift-native block and async support.